public/javascripts/login.js

```diff
@@ -1,54 +1,44 @@
-$(document).ready(function() {
-
+window.onload = function() {
   //start instance of app to connect to firebase
-  window.chubbo = new Chubbo();
-
-});
-
-
-function Chubbo() {
-
-  $('.cc-loginBtn').click(this.signIn.bind(this));
-  $('.cc-logoutBtn').click(this.signOut.bind(this));
+  window.login = new Login();
+}
 
+function Login() {
   this.initFirebase();
 }
 
-Chubbo.prototype.initFirebase = function() {
+Login.prototype.initFirebase = function() {
   this.auth = firebase.auth();
   this.auth.onAuthStateChanged(this.onAuthStateChanged.bind(this));
-};
+}
 
-Chubbo.prototype.signIn = function() {
+Login.prototype.signIn = function() {
   var provider = new firebase.auth.GoogleAuthProvider();
-  this.auth.signInWithPopup(provider).then(function(result) {
-    var token = result.credential.accessToken;
-    var user = result.user;
-    window.location = '/dashboard';
-    console.log(user.displayName);
+  //return promise to caller (navbar 'login' button)
+  return this.auth.signInWithPopup(provider).then(function(result) {
+    console.log(result.user.displayName);
   }).catch(function(error) {
     console.log('error logging in:', error.code);
   });
 }
 
-Chubbo.prototype.signOut = function() {
-  this.auth.signOut().then(function() {
-    window.location = '/';    
+Login.prototype.signOut = function() {
+  //return promise to caller (navbar 'logout' button)
+  return this.auth.signOut().then(function() {
   }).catch(function(error) {
     console.log('error logging out:', error.code);
   });
 }
 
-Chubbo.prototype.onAuthStateChanged = function(user) {
+Login.prototype.onAuthStateChanged = function(user) {
   if (user) {
-    toggleLoginBtn(true);
+    this.toggleLoginBtn(true);
   } else {
-    toggleLoginBtn(false);
+    this.toggleLoginBtn(false);
   }
 }
 
-
-function toggleLoginBtn(isLoggedIn) {
+Login.prototype.toggleLoginBtn = function(isLoggedIn) {
```

Comment:
it's called when the page first loads, to check whether or not a user is already logged in